index.html

```diff
@@ -2042,46 +1899,34 @@ <h3 id="applying">
         <h3>
           <code>display</code> member
         </h3>
+        <pre class="idl">
+          enum DisplayModeType {
+            "fullscreen",
+            "standalone",
+            "minimal-ui",
+            "browser"
+          };
+        </pre>
         <p>
-          The <dfn id="member-display"><code>display</code> member</dfn> is a
-          <a>string</a>, whose value is one of <a>display modes values</a>. The
-          item represents the developer's preferred <a>display mode</a> for the
-          web application. When the member is missing or erroneous, the user
-          agent MUST use the <a>fallback display mode</a>.
+          The <dfn>display</dfn> member is a <a>DisplayModeType</a>, whose
+          value is one of <a>display modes values</a>. The item represents the
+          developer's preferred <a>display mode</a> for the web application.
+          When the member is missing or erroneous, the user agent MUST use the
+          <a>fallback display mode</a>.
         </p>
         <p>
-          The <dfn>steps for processing the <code>display</code> member</dfn>
-          are given by the following algorithm. The algorithm takes a manifest
-          <var>manifest</var> as an argument, and returns a string.
+          The <dfn>steps for post-processing the <code>display</code>
+          member</dfn> are given by the following algorithm. The algorithm
+          takes a <a>WebAppManifest</a> <var>manifest</var> as an argument, and
+          returns a <a>DisplayModeType</a>.
         </p>
         <ol>
-          <li>Let <var>value</var> be the result of calling the
-          <a>[[\GetOwnProperty]]</a> internal method of <var>manifest</var>
-          passing "<code>display</code>" as the argument.
+          <li>Let <var>V</var> be <var>manifest["display"]</var>.
           </li>
-          <li>If <a>Type</a>(<var>value</var>) is not "string":
-            <ol>
-              <li>If <a>Type</a>(<var>value</var>) is not "undefined", <a>issue
-              a developer warning</a> that the type is unsupported.
-              </li>
-              <li>Return the <a>fallback display mode</a>'s value.
-              </li>
-            </ol>
-          </li>
-          <li>Otherwise, <a>Trim</a>(<var>value</var>), convert to lower-case,
-          and set <var>value</var> to be the result.
+          <li>If <a>Type</a>(<var>V</var>) is <code>null</code> return the <a>
+            fallback display mode</a>'s value.
```

Comment:
I'm confused about this (both the old and the new text, but since we're changing it, let's consider it): I think possibly this text has always been wrong, and should've said "default display mode", not "fallback display mode". Each display mode has a defined "fallback display mode" (e.g., "fullscreen"'s FDM is "standalone"), so it doesn't make sense to say "the fallback display mode" when it's undefined.

Rather it should say "the default display mode", but since that is defined as "browser", why not just say "browser"? And if you're going to say that, why not just set a default of "browser" in the IDL, and delete this entire post-processing step?

Reply:
Agree! We get "browser" for free now 🎉

Reply:
Well, the idea is that if a browser doesnt support "fullscreen" but "standalone" then it should render "fullscreen" as "standalone" and not "browser" - that is how it was implemented in Chrome when it didnt support fullscreen

Reply:
ok followed advice

Reply:
> Well, the idea is that if a browser doesnt support "fullscreen" but "standalone" then it should render "fullscreen" as "standalone" and not "browser"

Yeah. Looks good now. To be clear, I'm not proposing we remove the concept of FDM --- that is good that it works like you describe. Just saying that there is no "global fallback display mode" (rather, that each display mode has its own FDM). It's more of a default than a fallback.

No further action required.